example/node_classification.py

```diff
@@ -0,0 +1,237 @@
+from torch_geometric.nn import GCNConv
+import torch.nn.functional as F
+from torch_geometric.datasets import Planetoid
+from torch_geometric.transforms import NormalizeFeatures
+import mgp
+import numpy as np
+from tqdm import tqdm
+import random
+import typing
+
+TORCH_SEED = 1234567
+NO_CLASSES = 7
+NO_FEATURES = 1433
+
+HIDDEN_CHANNELS = 16
+LEARNING_RATE = 0.01
+WEIGHT_DECAY = 5e-4
+
+
+class Data(object):
+    def __init__(
+        self, x: torch.tensor, 
+        y: torch.tensor, 
+        edge_index: torch.tensor, 
+        train_mask: torch.tensor, 
+        val_mask: torch.tensor, 
+        test_mask: torch.tensor):
+
+        self.x = x
+        self.y = y
+        self.edge_index = edge_index
+        self.train_mask = train_mask
+        self.val_mask = val_mask
+        self.test_mask = test_mask
+
+class GCN(torch.nn.Module):
+    def __init__(self, hidden_channels):
+        super().__init__()
+        torch.manual_seed(TORCH_SEED)
+        self.conv1 = GCNConv(NO_FEATURES, hidden_channels)
+        self.conv2 = GCNConv(hidden_channels, NO_CLASSES)
+        
+
+    def forward(self, x, edge_index):
+        x = self.conv1(x, edge_index)
+        x = x.relu()
+        x = F.dropout(x, p=0.5, training=self.training)
+        x = self.conv2(x, edge_index)
+        return x
+
+
+
+
+
+def train_model() -> torch.tensor:
+    model.train()
+    optimizer.zero_grad()  # Clear gradients.
+    out = model(data.x, data.edge_index)  # Perform a single forward pass.
+    loss = criterion(out[data.train_mask], data.y[data.train_mask])  # Compute the loss solely based on the training nodes.
+    loss.backward()  # Derive gradients.
+    optimizer.step()  # Update parameters based on gradients.
+    return loss
+
+
+
+def accuracy(mask) -> float:
+    model.eval()
+    out = model(data.x, data.edge_index)
+    pred = out.argmax(dim=1)  # Use the class with highest probability.
+    correct = pred[mask] == data.y[mask]  # Check against ground-truth labels.
+    acc = int(correct.sum()) / int(mask.sum())  # Derive ratio of correct predictions.
+    return acc
+
+def convert_data(
+    nodes: typing.List[mgp.Vertex], 
+    edges: typing.List[mgp.Edge], 
+    train_ratio: float, 
+    val_ratio: float, 
+    test_ratio: float)->Data:
+    
+    assert len(nodes) > 0, "dataset is empty"
+    
+    x = np.zeros((len(nodes),len(nodes[0].properties.get("features"))))
+    y = np.zeros((len(nodes)))
+    edge_index = np.zeros((2,len(edges)))
+    train_mask = np.zeros((len(nodes)))
+    val_mask = np.zeros((len(nodes)))
+    test_mask = np.zeros((len(nodes)))
+    
+    masks = np.zeros((len(nodes)))
+    
+    
+    # reindexing = {} # for unordered ids
+    # for i in range(len(nodes)): 
+    #     reindexing[nodes[i].properties.get("id")] = i
+
+
+    assert train_ratio > 0 and val_ratio > 0 and test_ratio > 0, "ratios are positive numbers in [0,1]"
+    assert train_ratio + val_ratio + test_ratio == 1, "data ratios don't add up to 100%"
```

Comment:
This should be handled by throwing an exception.

Reply:
Correct me if I'm wrong, assert checks whether condition is satisfied, then throws exception with the string after comma? I thought strings will be shown assert is not satisfied.

Reply:
I understand now. Repaired.